text/0298-deprecate-component-lifecycle-events.md

```diff
@@ -0,0 +1,103 @@
+- Start Date: 2018-01-19
+- RFC PR: https://github.com/emberjs/rfcs/pull/298
+- Ember Issue: (leave this empty)
+
+# Deprecate Component Lifecycle Events
+
+## Summary
+
+The goal of this RFC is to remove component lifecycle events, such as `Ember.on('didRender'`,
+in favor of enforcing usage of the lifecycle methods themselves i.e. `didRender(){}`
+
+## Motivation
+
+When using events, timing is not guaranteed, so it is possible to have unintended
+consequences and have things execute in an unexpected order. Using the methods directly
+does guarantee timing and solidifies the path forward on one recommended way of using
+lifecycle hooks.
+
+## Detailed design
+
+Ember will start logging deprecation messages that state which event you used and which
```

Comment:
AFAICT the RFC only gives examples of the events we are talking about, but the detailed design section should also include an explicit list of events.

I _believe_ this is the current list of events that are component specific ([from the curly component manager](https://github.com/emberjs/ember.js/blob/7ff756a87d5757d05588136d41516bcd5f5a0fd1/packages/ember-glimmer/lib/component-managers/curly.ts)):

* `willRender`
* `willInsertElement`
* `didInsertElement`
* `didRender`
* `didUpdateAttrs`
* `didReceiveAttrs`
* `willUpdate`
* `willRender`
* `didUpdate`
* `didRender`